common/file/archive/zip.go

```diff
@@ -0,0 +1,178 @@
+package archive
+
+import (
+	"archive/zip"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	log "github.com/thrasher-corp/gocryptotrader/logger"
+)
+
+const (
+	// ErrUnableToCloseFile message to display when file handler is unable to be closed normally
+	ErrUnableToCloseFile string = "Unable to close file %v %v"
+)
+
+var (
+	addFilesToZip func(z *zip.Writer, src string, isDir bool) error
+)
+
+func init() {
+	addFilesToZip = addFilesToZipWrapper
+}
+
+// UnZip extracts input zip into dest path
+func UnZip(src, dest string) (fileList []string, err error) {
+	z, err := zip.OpenReader(src)
+	if err != nil {
+		return
+	}
+
+	for x := range z.File {
+		fPath := filepath.Join(dest, z.File[x].Name) // nolint:gosec
+		// We ignore gosec linter above because the code below files the file traversal bug when extracting archives
+		if !strings.HasPrefix(fPath, filepath.Clean(dest)+string(os.PathSeparator)) {
+			err = z.Close()
+			if err != nil {
+				log.Errorf(log.Global, ErrUnableToCloseFile, z, err)
+			}
+			err = fmt.Errorf("%s: illegal file path", fPath)
+			return
+		}
+
+		if z.File[x].FileInfo().IsDir() {
+			err = os.MkdirAll(fPath, os.ModePerm)
+			if err != nil {
+				return
+			}
+			continue
+		}
+
+		err = os.MkdirAll(filepath.Dir(fPath), 0770)
+		if err != nil {
+			return
+		}
+
+		var outFile *os.File
+		outFile, err = os.OpenFile(fPath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, z.File[x].Mode())
+		if err != nil {
+			return
+		}
+
+		var eFile io.ReadCloser
+		eFile, err = z.File[x].Open()
+		if err != nil {
+			err = outFile.Close()
```

Comment:
This is real nit picky but if this actually does fail it will override the err above and you will never know what happened. Maybe just have this outFile.Close() as a separate error var and all close methods separate as well?

Reply:
Yep should have allocated to new var to allow error to be captured good pick up fixed :D